middlewares/compress.go

```diff
@@ -6,12 +6,25 @@ import (
 	"github.com/NYTimes/gziphandler"
 )
 
-// Compress is a middleware that allows redirections
-type Compress struct {
-}
+const (
+	contentEncodingHeader = "Content-Encoding"
+)
+
+// Compress is a middleware that allows redirection
+type Compress struct{}
 
-// ServerHTTP is a function used by negroni
+// ServerHTTP is a function used by Negroni
 func (c *Compress) ServeHTTP(rw http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
-	newGzipHandler := gziphandler.GzipHandler(next)
-	newGzipHandler.ServeHTTP(rw, r)
+	if isEncoded(r.Header) {
+		next.ServeHTTP(rw, r)
+	} else {
+		newGzipHandler := gziphandler.GzipHandler(next)
+		newGzipHandler.ServeHTTP(rw, r)
+	}
+}
+
+func isEncoded(headers http.Header) bool {
+	header := headers.Get(contentEncodingHeader)
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Encoding
```

Comment:
Why do we add this link here? I suppose it's to reference what constitutes an encoding?

If so, could we add an explaining comment? Something like "According to http://..., content is encoded if ...".